pkg/deploy/generator/resources_rp.go

```diff
@@ -476,6 +476,7 @@ func (g *generator) rpVMSS() *arm.Resource {
 		// TODO: Replace with Live Service Configuration in KeyVault
 		"clustersInstallViaHive",
 		"clusterDefaultInstallerPullspec",
+		"clustersAdoptByHive",
```

Comment:
Having `clustersInstallViaHive` and `clustersAdoptByHive` in this array causes the following ARM template error:

`Unable to process template language expressions for resource 'xxxxxxxxx' at line '1' and column '7182'. 'The template language function 'base64' expects its parameter to be of type 'String'`
 
 To get around this, I moved `clustersInstallViaHive` and `clustersAdoptByHive` to the array on line 489 - the one with `adminApiCaBundle` and `armApiCaBundle` - so `base64` does not get called on these boolean parameters.

Reply:
To get around a separate issue with the aro rp service not starting, I reverted this change and tried making these fields strings in the ARM template. This change also allowed the ARM template validation to succeed, but the aro-rp service would not start.

Reply:
This should have been fixed in the last commit by @hawkowl https://github.com/Azure/ARO-RP/pull/2380/commits/1bbc858a98a1f051a9980fc98cced5ad2abdee63